magento/magento2
# Set billing address only if different in checkout

### Description
Reduce duplicate calls to modify the billing address if it already matches the 'new' value. This prevents an infinite loop caused by some address observers.

### Related Pull Requests
*None*

### Fixed Issues
*None*

### Manual testing scenarios
1. In checkout, observe that "quote.billingAddress()" gets called with the same address multiple times before this change, and only once after.

1. Install `clearpay/module-clearpay`
1. Configure reward points or coupon code to obtain zero-price checkout
1. Ensure zero subtotal payment method is enabled
1. Use reward points or coupon code to reduce cart price to zero
1. Observe infinite loop/spinner without this change, and normal checkout experience with.

### Questions or comments
I have not run any tests although I expect none to fail because of this change.

### Contribution checklist (*)

### Resolved issues:
1. [x] resolves magento/magento2#30553: Set billing address only if different in checkout
